server/aws-lsp-codewhisperer/src/shared/amazonQServiceManager/AmazonQTokenServiceManager.ts

```
@@ -318,21 +320,21 @@ export class AmazonQTokenServiceManager extends BaseAmazonQServiceManager<
         }
 
         const parsedArn = parse(newProfileArn)
-        const endpoint = AWS_Q_ENDPOINTS.get(parsedArn.region)
+        const region = parsedArn.region
+        const endpoint = AWS_Q_ENDPOINTS.get(region)
         if (!endpoint) {
             throw new Error('Requested profileArn region is not supported')
         }
 
-        const profiles = await getListAllAvailableProfilesHandler(this.serviceFactory)({
-            connectionType: 'identityCenter',
-            logging: this.logging,
-            token: token,
-            endpoints: new Map([[parsedArn.region, endpoint]]),
-        })
-
-        this.handleTokenCancellationRequest(token)
-
-        const newProfile = profiles.find(el => el.arn === newProfileArn)
+        // Hack to inject a dummy profile name as it's not used by client IDE for now, if client IDE starts consuming name field then we should also pass both profile name and arn from the IDE
+        // When service is ready to take more tps, revert https://github.com/aws/language-servers/pull/1329 to add profile validation
+        const newProfile: AmazonQDeveloperProfile = {
```

When we accept invalid profiles, we need to make sure that when service calls fail due to invalid profile, corresponding error is surfaced to extension from user-facing features. 
Otherwise, we'll show generic errors to customers without a way to recover.

Proper UX would be to return `E_AMAZON_Q_INVALID_PROFILE` LSP response error code to extension, and extension triggers profile selection UX.

---

yep we aligned it's a followup to surface proper error message